src/eos-crash-metrics.c

```diff
@@ -0,0 +1,196 @@
+/* Copyright 2017 Endless Mobile, Inc.
+ *
+ * This file is part of eos-metrics-instrumentation.
+ *
+ * eos-metrics-instrumentation is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * eos-metrics-instrumentation is distributed in the hope that it will be
+ * useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with eos-metrics-instrumentation.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdlib.h>
+#include <eosmetrics/eosmetrics.h>
+#include <ostree.h>
+
+#define PROGRAM_DUMPED_CORE_EVENT "ed57b607-4a56-47f1-b1e4-5dc3e74335ec"
+
+static GVariant *
+hashtable_to_variant (GHashTable *ht)
+{
+  g_autoptr(GList) keys = g_hash_table_get_keys (ht);
+  g_auto(GVariantBuilder) builder = G_VARIANT_BUILDER_INIT (G_VARIANT_TYPE ("a{sv}"));
+
+  g_variant_builder_init (&builder, G_VARIANT_TYPE ("a{sv}"));
+
+  GList *key_iter = keys;
+  for (; key_iter; key_iter = key_iter->next)
+    {
+      g_variant_builder_open (&builder, G_VARIANT_TYPE ("{sv}"));
+
+      g_variant_builder_add (&builder, "s", g_strdup (key_iter->data));
+      g_variant_builder_add (&builder, "v", g_variant_ref (g_hash_table_lookup (ht, key_iter->data)));
+
+      g_variant_builder_close (&builder);
+    }
+
+  return g_variant_builder_end (&builder);
+}
+
+static void
+report_crash (const char *binary,
+              gint16 signal,
+              gint64 timestamp,
+              const char *ostree_commit,
+              const char *ostree_url)
+{
+  GVariantDict dict;
+  g_variant_dict_init (&dict, NULL);
+
+  g_variant_dict_insert (&dict, "binary", "v", g_variant_new_string (binary));
```

Comment:
You either want to use `g_variant_dict_insert_value()` which takes a `GVariant` or use `g_variant_insert(&dict, $key, $type, $value)`

Reply:
Makes sense